Group node fixes

martint-unity opened the pull request:
### Purpose of this PR
Fixes for these fog bugzzzz

Sticky note not being grouped properly (FB-1181447)
Before:
![group_sticky_note_broken](https://user-images.githubusercontent.com/35328557/65022548-78833380-d931-11e9-8cb1-df8a8ea36f8f.gif)

After:
![group_sticky_note](https://user-images.githubusercontent.com/35328557/65021470-77510700-d92f-11e9-8557-c574d7ef78f4.gif)


Coy nodes from a group causes errors (FB-1182137)
Before:
![copy_node_from_group_broken](https://user-images.githubusercontent.com/35328557/65022393-322dd480-d931-11e9-8488-7baae4a8b8c9.gif)

After:
![copy_node_from_group](https://user-images.githubusercontent.com/35328557/65021306-1a555100-d92f-11e9-85f0-66675a4f4d10.gif)


---
### Testing status
**Katana Tests**: First off we need to make sure the Katana SRP tests are green?


Alternative, launch Yamato (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/sg%252Fui-ux-group-node-fixes/.yamato%252Fupm-ci-shadergraph.yml%2523All_ShaderGraph/468503/job

---
### Overall Product Risks
**Technical Risk**: None, Low, Medium, High?
Low

**Halo Effect**: None, Low, Medium, High?
Low

---

phi-lira commented on com.unity.shadergraph/CHANGELOG.md:
```diff
@@ -10,6 +10,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - You can now save your graph as a new Asset.
 
+### Fixed
+- Sticky Notes can now be grouped properly.
```
Please link the public bug from issue tracker

Kink3d replied:
This is not a convention we follow (at this time). Why is this a valid reason to deny a PR?

martint-unity commented:
@zadi0648 @pbbastian 
Anything you guys wants to add? :)

martint-unity commented:
@zadi0648 Updated with your feedback

zadi0648 commented:
Why make ungroup the last contextual menu item? Menu items of the same kind should be together ya?
<img width="243" alt="GroupNodeFixes0" src="https://user-images.githubusercontent.com/3393041/66165658-15102b80-e5ea-11e9-9e2c-161260d02271.png">

Instead I'd recommend just giving the sticky note the Group Option, for parity/consistency, and keeping Ungroup right under that
<img width="253" alt="GroupNodeFixes1" src="https://user-images.githubusercontent.com/3393041/66165696-2c4f1900-e5ea-11e9-8904-fda56e9b0381.png">

martint-unity commented:
I totally agree! Done :)

zadi0648 commented:
Improved!

Minor parity thing, but the option to group a single node/sticky note is greyed out for a sticky note, while not so for main nodes:

<img width="328" alt="Grouping0" src="https://user-images.githubusercontent.com/3393041/66524166-7f234780-eaa6-11e9-801e-39cafe99a07f.png">
<img width="328" alt="Grouping1" src="https://user-images.githubusercontent.com/3393041/66524184-89dddc80-eaa6-11e9-8330-3fcc712e8635.png">

Also, as for the `Delete Group and Contents` item, could we perhaps improve that some more in this PR? Suggesting:

- Add a separator between it and the other stuff (Cut/Copy)
- Add another option with it above `Delete Group` (that only deletes the group, not the contents - follows the rule that every hotkey should come with a menu option somewhere)
- It doesn't show up when the right click happens to be on Nodes or Sticky Notes (see 2nd image). Perhaps it should remain that way? I'll leave that to your discretion.
- Perhaps the grouping stuff should remain together (despite it being greyed out at the bottom)

<img width="435" alt="Grouping2" src="https://user-images.githubusercontent.com/3393041/66524223-a2e68d80-eaa6-11e9-84a0-2822eb65fc7e.png">
<img width="435" alt="Grouping3" src="https://user-images.githubusercontent.com/3393041/66524224-a548e780-eaa6-11e9-930e-5b3dd9668f39.png">

zadi0648 commented:
Actually I find that Nodes are inconsistent of when you can and can't group them (when they are alone), do I make this a new fogbugz?

Repro:
1) Left Click GV Background
2) Right Click on Node
> Option to Group Node is available
3) Left Click GV Background
4) Right Click GV Background
5) Right Click on Node
> Option to Group Node is unavailable

![CantGroup0](https://user-images.githubusercontent.com/3393041/66528497-f069f700-eab4-11e9-835f-e97648c7b172.gif)

martint-unity commented:
Fixed the above mentioned issue.
Also added nicer menus.
Groups are also a bit transparent so that you can see if another group is under it.
Also added so that you can select a group "behind" another group and move it.

martint-unity commented:
![2019-10-22_10-30-19](https://user-images.githubusercontent.com/35328557/67270192-76743d80-f4b8-11e9-8d77-d853ae4a6255.gif)

martint-unity commented:
I also added shortcuts to the menu. Should be OS agnostic.

martint-unity commented:
This fails due to Adds or something @sophiaaar Knows more about it. But Master has the same issues for the last 6 days so Nothing this branch introduced so this should be safe to push @pbbastian
